tests/sqlparser_bigquery.rs

```diff
@@ -2214,6 +2214,30 @@ fn test_select_as_value() {
     assert_eq!(Some(ValueTableMode::AsValue), select.value_table_mode);
 }
 
+#[test]
+fn test_typed_strings() {
+    let expr = bigquery().verified_expr(r#"JSON """{"foo":"bar's"}""""#);
+    assert_eq!(
+        Expr::TypedString {
+            data_type: DataType::JSON,
+            value: Value::TripleDoubleQuotedString(r#"{"foo":"bar's"}"#.into())
+        },
+        expr
+    );
+
+    let expr = bigquery().verified_expr(r#"JSON '''{"foo":"bar's"}'''"#);
+    if let Expr::TypedString { data_type, value } = expr {
+        assert_eq!(DataType::JSON, data_type);
+        let string_value: String = value.into();
+        assert_eq!(r#"{"foo":"bar's"}"#, string_value);
+    }
+
+    // SingleQuotedString and DoubleQuotedString are currently not correctly formatted by `fmt::Display for Value`.
+    // BigQuery does not support double escaping, should be \' or \" instead.
+    //bigquery().verified_expr(r#"JSON '{"foo":"bar\'s"}'"#);
+    //bigquery().verified_expr(r#"JSON "{\"foo\":\"bar's\"}""#);
```

Comment:
is this part a TODO for this PR?

Reply:
No, like I mentioned in the description, this would be better for another PR. I can remove those lines if you think that's better but I thought it's good to document this unintended behavior. (The test should be there, it's just failing at the moment.)

Reply:
The issue I ran into is this SQL `SELECT JSON """{"foo":"bar's"}"""` gets parsed and then re-printed as `SELECT JSON '{"foo":"bar's"}'` (note the not escaped `'`). But let me do some more tests – for now I'll take this out of this PR

Reply:
I think we shouldnt need to set up unescape or other special modes to handle this case. The parser ingests valid input SQL and outputs buggy SQL in the case that @graup brings up so it's a pretty clear cut case of a bug?